modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc

```diff
@@ -303,17 +304,20 @@ CONCAT_WS(''," . $DB->quote($dir_path) . ",'trashbin/',SUBSTRING_INDEX(minc_loca
             'minYear'         => $study['startYear'] - $study['ageMax'],
             'maxYear'         => $study['endYear'] - $study['ageMin']
         );
-        $problemOptions = array(
-            '' => 'Any',
-            'CandID and PSCID do not match database'
-            => 'Candidate Mismatch',
-            'Could not identify scan type'
-            => 'Could not identify scan type',
-            'Protocol Violation'
-            => 'MRI Protocol Check violation',
-            'Visit label does not exist'
-            => 'Visit label does not exist'
+
+        $problemOptions = ['' => 'Any'];
+
+        // Build a list of existing problem types based on data in db
+        $problemTypes = $db->pselect(
+            "SELECT DISTINCT MRICandidateErrors.Reason FROM MRICandidateErrors
+            UNION 
+            SELECT DISTINCT 'Could not identify scan type' FROM mri_protocol_violated_scans
```

Comment:
@driusan 
1. `SELECT DISTINCT MRICandidateErrors.Reason` adds all the reasons from `MRICandidateErrors` to the dropdown
2. `SELECT DISTINCT 'Could not identify scan type'` adds an extra option if there's at least one record in `mri_protocol_violated_scans`
3. `SELECT DISTINCT 'Protocol Violation'` add an extra option if there's at least one record in `mri_violations_log`

That is the behaviour that @MounaSafiHarab asked for